tycho-plugins/repository-utils/src/main/java/org/jboss/tools/tycho/sitegenerator/FetchSourcesFromManifests.java

```diff
@@ -351,6 +368,75 @@ public void execute() throws MojoExecutionException, MojoFailureException {
 		// getLog().info("Written to " + digestFile.toString() + ":\n\n" + sb.toString());
 	}
 
+	/*
+	 * Given a set of zip files, unpack them and merge them into a single combined source zip
+	 * If mode == PURGE_ZIPS, delete zips to save disk space, keeping only the combined zip
+	 * If mode == CACHE_ZIPS, move zips into cache folder
+	 * 
+	 */
+	private void createCombinedZipFile(Set<File> zipFiles, int mode)
```

Comment:
How would that be done? Please throw me some code, not just a cryptic suggestion. My mojo fu is not as strong as yours. :(